src/map/trade.cpp

```diff
@@ -413,10 +413,23 @@ void trade_tradeadditem(map_session_data *sd, int16 index, int16 amount)
 	// Locate a trade position
 	ARR_FIND( 0, 10, trade_i, sd->deal.item[trade_i].index == index || sd->deal.item[trade_i].amount == 0 );
 	if( trade_i == 10 ) { // No space left
-		// clif_tradeitemok(*sd, index, EXITEM_ADD_FAILED_OVERWEIGHT); // We do not know if the server respond with this or not since the official client prevents this case client-side.
 		return;
 	}
 
+	// Fail to add the item if the inventory is full
+	if (pc_inventoryblank(target_sd) <= trade_i) {
```

Comment:
Should this be moved into line 446?
"New deal item"?

Reply:
No, on officials this is checked each time you add an item, and before it's added to sd->deal.item (otherwise it will return the item, basically duplicating them)